src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -9804,6 +9808,84 @@ HRESULT ProfToEEInterfaceImpl::ApplyMetaData(
     return hr;
 }
 
+// Add path to list of assemblies used by the runtime to probe assembly references
+// pAssemblyPath should be absolute path of assembly
+// Assembly name should match filename.
+HRESULT ProfToEEInterfaceImpl::AddAssemblyPath(AppDomainID appDomainId,
+    const WCHAR *pAssemblyPath)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        // Access to TPA list require lock
+        CAN_TAKE_LOCK;
+        SO_NOT_MAINLINE;
+    }
+    CONTRACTL_END;
+
+    PROFILER_TO_CLR_ENTRYPOINT_SYNC_EX(kP2EEAllowableAfterAttach, (LF_CORPROF, LL_INFO1000, "**PROF: AddAssemblyPath.\n"));
+
+    BaseDomain* pDomain = reinterpret_cast<BaseDomain*>(appDomainId);
+    if (pDomain == NULL)
+    {
+        return CORPROF_E_DATAINCOMPLETE;
```

Comment:
```suggestion
        return E_INVALIDARG;
```

We typically use CORPROF_E_DATA_INCOMPLETE to refer to runtime data structures that aren't fully initialized or have already begun to unload. In this case the caller just gave us a bad AppDomainID